src/datatable.py

```diff
@@ -0,0 +1,192 @@
+        output = {}
+        output['draw'] = str(int(self.request['draw']))
+        output['recordsTotal'] = str(self.records_total)
+        output['recordsFiltered'] = str(self.records_filtered)
+
+        data_rows = []
+        i = self.paging().start
+        for row in self.result_data:
+            data_row = []
+            i += 1
+            data_row.append(i)
+            for col, val in row.items():
+                uwu = ""
+                if col == "_id":
+                    if self.group_by == "fullgnomen":
+                        uwu = '<a href="/variant/{0}">{0}</a>'.format(val)
+                    elif self.group_by == "dn_no":
+                        uwu = '<a href="/patient/{0}">{0}</a>'.format(val)
+                    elif self.group_by == "gene":
+                        uwu = '<a href="/gene/{0}">{0}</a>'.format(val)
+                elif col == "total":
+                    uwu = val
+                else:
+                    # filtering like this doesn't work yet
+                    # uwu = '<a href="/all?{1}={0}">{0}</a>'.format(val, col)
+                    uwu = val
+                data_row.append(uwu)
+            data_rows.append(data_row)
+
+        output['data'] = data_rows
+        return output
+
+    def run_queries(self):
+        pages = self.paging()  # pages has 'start' and 'length' attributes
+        filter = self.filtering()  # the term entered into the datatable search
+        sorting = self.sorting()  # the document field chosen to sort
+
+        # get result from db
+        match = {"$match": filter}
+        group = {}
+        if self.group_by is not None:
+            group = {
+                "$group": {
+                    "_id": "$" + self.group_by,
+                    "total": {
+                        "$sum": 1
+                    }
+                }
+            }
+
+        if self.group_by == "fullgnomen":
+            group["$group"]["protein"] = {"$addToSet": "$pnomen"}
+        pipeline = []
+
+        if group:
+            pipeline.append(group)
+        if filter:
+            pipeline.append(match)
+        pipeline = pipeline + [{"$sort": sorting}, {"$skip": pages.start}]
+        if pages.length >= 0:
+            pipeline.append({"$limit": pages.length})
+        self.result_data = list(self.dbh[self.collection].aggregate(pipeline))
+
+        if group:
+            # total amount unfiltered & total amount filtered (search bar)
+            self.records_total = len(list(self.dbh[self.collection].aggregate([group])))
+            self.records_filtered = len(list(self.dbh[self.collection].aggregate([group, match])))
+        else:
+            self.records_total = len(list(self.dbh[self.collection].find()))
+            self.records_filtered = len(list(self.dbh[self.collection].find(filter)))
+
+    def filtering(self):
+        """
+        build your filter spec
+        "search": {
+            "regex": "false",
+            "value": ""
+        },
```

Comment:
Could you explain these lines? I don't understand these comments. I cannot find any variable with search, neither as input to the function or output.

Reply:
see earlier comment:
I have made another issue #19, so that the next version needs to have a look.

Reply:
I think this was left by the developer to help understand the difference between python and mongodb. But because this can be looked up in the mongodb documentation I removed them.